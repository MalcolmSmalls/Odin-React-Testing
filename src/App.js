// import './App.css';

// function App() {
//   return (
//     <div className="App">
//       <h1>Our First Test</h1>
//     </div>
//   );
// }

// export default App;



import React from "react"

export default function App () {
  const [ title, setTitle ] = React.useState('Magnificent Monkeys')

  function handleClick () {
    setTitle('Radical Rhinos')
  }

  return (

    <div>
      <h1>{title}</h1>
      <button onClick={handleClick}>Change title</button>
    </div>

  )
}