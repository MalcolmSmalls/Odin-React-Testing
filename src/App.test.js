import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

describe("App component", () => {
    it("renders magnificent monkeys", () => {
        const { container } = render(<App />);
        expect(container).toMatchSnapshot()
    });

    it("renders radical rhinos after click", () => {
        render(<App />);
        const button = screen.getByRole("button", { name: "Change title"});

        userEvent.click(button);

        expect(screen.getByRole("heading").textContent).toMatch(/radical rhinos/i)


    })
});