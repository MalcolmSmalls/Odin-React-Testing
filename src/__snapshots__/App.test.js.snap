// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders magnificent monkeys 1`] = `
<div>
  <div>
    <h1>
      Magnificent Monkeys
    </h1>
    <button>
      Change title
    </button>
  </div>
</div>
`;
